js/privatebin.js

```diff
@@ -2295,6 +2129,192 @@ jQuery.PrivateBin = function($, sjcl, Base64, RawDeflate) {
             I18n._(me.$attachmentLink, label, me.$attachmentLink.attr('download'));
         }
 
+
+        /*
+         * Read file data as dataURL using the FileReader API
+         * https://developer.mozilla.org/en-US/docs/Web/API/FileReader#readAsDataURL()
+         */
+        me.readFileData = function (file) {
+            if (typeof FileReader === 'undefined') {
+                // revert loading status…
+                AttachmentViewer.hideAttachment();
+                AttachmentViewer.hideAttachmentPreview();
+                Alert.showError(I18n._('Your browser does not support uploading encrypted files. Please use a newer browser.'));
+                return;
+            }
+
+            var fileReader = new FileReader();
+            if (file === undefined) {
+                file = me.$fileInput[0].files[0];
+                $('#dragAndDropFileName').text('');
+            } else {
+                $('#dragAndDropFileName').text(file.name);
+            }
+
+            me.file = file;
+
+            fileReader.onload = function (event) {
+                var dataURL = event.target.result;
+                me.attachmentData = dataURL;
+
+                if (Editor.isPreview()) {
+                    me.handleAttachmentPreview(AttachmentViewer.$attachmentPreview, dataURL);
```

Comment:
If you are in `AttachmentViewer` you can just use `me` instead of referring to it with its full name.